### PR TITLE
allow a command-specified project to be selected during init

### PR DESCRIPTION
**Pull request by bkendall**
### Description

When running `firebase init --project <project>`, the init flow will now attempt to use the provided project. If that project does not exist, the command fails. If a project is omitted, the normal prompt flow appears.
	 
### Scenarios Tested

* init with a project that exists
* init with a project that doesn't exist
* init without a project

### CODE REVIEW

**Comment by samtstern** [comment]
@bkendall does this affect https://github.com/firebase/firebase-tools/issues/1761 at all?

File: src/init/features/project.ts [hunk 1]
```diff
@@ -32,6 +32,11 @@ export interface ProjectInfo {
   location?: string; // maps to FirebaseProjectMetadata.resources.locationId
 }
 
+/**
```
Comment (samtstern) [inline src/init/features/project.ts]: wait so we're doing `jsdoc` now?  IMO the comment "Transforms project metadata to a ProjectInfo object." is useful but the rest is just a restatement of the type signature.

File: mocha.opts [hunk 2]
```diff
@@ -3,4 +3,4 @@
 --require src/test/helpers/mocha-bootstrap.js
 --recursive
 --timeout=1000
-src/test/**/*.{ts,js}
+src/test/init/features/project.spec.ts
```
Comment (samtstern) [inline mocha.opts]: Remove :-)

**Comment by bkendall** [comment]
@samtstern This does fix #1761

File: src/init/features/project.ts (hunk 1 above)
Comment (bkendall) [reply src/init/features/project.ts]: I want to turn down this rule to "doc only exported functions"... let me dig for a second...

Comment (bkendall) [reply src/init/features/project.ts]: Oh yeah, I can do this. Winning.

**Pull request merged**